### ISSUE TITLE
[Serious] Keyboard focus order is not logical

### ISSUE THREAD

**Issue by ntallman** [APTrust/dart#594]
## Issue Details

| | |
|---|---|
| **Source Issue ID** | 87754 |
| **Checkpoint** | 2.4 Navigable › Focus Order (2.4.3.a) |
| **Impact** | Serious |
| **Standards** | wcag2a, wcag2aa, wcag21a, wcag21aa, wcag22a, wcag22aa |
| **Test Area** | 02 Import/Export Workflow |
| **URL** | <https://aptrust.org/documentation-page/dart-digital-archivists-resource-tool/> |
| **Environment** | Production |
| **Platform** | Windows | Chrome | Firefox |
| **Assistive Technology** | JAWS | NVDA | Magnifier |
| **Method** | Manual |
| **Digital Asset Type** | Desktop Web |
| **Release** | 4.2026 |
| **Date Reported** | 05/12/26 at 12:39 pm |
| **Issue Status** | Open |
| **Unit Type** | Page |
| **Issue Type** | Accessibility |
| **Reported By** | kim.casey@ablr360.com |

## Description

Keyboard focus order is not logical in such a way that it affects the meaning or operability of the content.
Pattern for confirmation dialogs: In Export Settings, select New, then navigate and add a couple of Setup Questions.  Then select one to delete.  In the confirmation dialog, the keyboard focus is placed on the OK button rather than the prompt. Also occurs in the confirm Delete dialog (select Delete button at bottom). See Import Export User Experience video at 12:35 and 17:05.

## Recommendation

Rule : 
The reading and navigation order MUST be logical and intuitive.

How To Fix: 
Fix this issue by ensuring the DOM order is logical so that the focus order will naturally follow. Do not use positive tabindex values to "correct" the tab order as it will not change the reading order for screen reader users and will almost always cause unintended, significant problems.

Reference: 
Accessible dialogs (when the dialog popup is not announced)
Accessible Dialog Modal example: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/
Background: 
When keyboard focusable components do not receive focus in a logical order, people with mobility impairments, reading disabilities, and low vision are all impacted. The keyboard focus order must be logical and consistent with the layout of the content. (Note: This does not mean that the focus order has to be identical to the visual order, as long as the user can still understand and operate the content.) A logical focus order makes interaction with content predictable for people who rely on a keyboard to interact with a web content.

## Screenshots

![02_focus logical1](https://raw.githubusercontent.com/APTrust/dart/master/docs/accessibility-audit/screenshots/87754/02_focus_logical1.png)
